weblab/entities/views.py

```diff
@@ -778,6 +779,42 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class RenameView(LoginRequiredMixin, UserPassesTestMixin, FormMixin, EntityTypeMixin, DetailView):
+    template_name = 'entities/entity_rename_form.html'
+    context_object_name = 'entity'
+    form_class = EntityRenameForm
+
+    def _get_object(self):
```

Comment:
The `DetailView` class sets `self.object` so we don't need this method.

Reply:
When I try and take this out I get an error message saying the method is required

Reply:
That might be just because it's used in the `post` and `test_func` methods below? In which case implementing the other changes would remove the need for it. Or it might be that one of the base classes needs it, in which case leave it in.